src/ui/display_write_video_thread.py

```diff
@@ -25,10 +26,11 @@ def run(self):
 
         self.model = Model(weights_path)
         self.Fourcc = cv2.VideoWriter_fourcc(*'mp4v')
-        self.Output = cv2.VideoWriter('video_recording.mp4', self.Fourcc, 20, (640, 480))
+        self.Output = cv2.VideoWriter('video_recording'+str(self.cameraNumber)+'.mp4', self.Fourcc, 20, (640, 480))
+
         #self.Capture = cv2.VideoCapture(0)
 
-        while self.ThreadActive: 
+        while True: 
             ret, frame = Capture.read() #get frame from video feed
             if ret: 
                 frame = cv2.cvtColor(frame, cv2.COLOR_BGR2RGB) #get color image from feed
```

Comment:
Possible solution: cv2 normally expects BGR for frames. We are converting from BGR to RGB here. Try converting it back to BGR after the model has added its annotations and see if it can write again.

Reply:
That worked, but the video is still sped up for some reason.